packages/core/vue/src/useWallet.ts

```diff
@@ -0,0 +1,225 @@
+} from '@solana/wallet-adapter-base';
+import { Wallet, WalletName } from '@solana/wallet-adapter-wallets';
+import { useLocalStorage } from './useLocalStorage';
+import { WalletNotSelectedError, OperationNotSupportedByWalletError } from './errors';
+
+type Adapter = ReturnType<Wallet['adapter']>;
+type WalletDictionary = { [key: string]: Wallet };
+
+export interface WalletStore {
+    // Props.
+    wallets: Wallet[];
+    autoConnect: boolean;
+
+    // Data.
+    walletName: Ref<string | null>,
+    walletsByName: Ref<WalletDictionary>,
+    wallet: Ref<Wallet | null>;
+    adapter: Ref<Adapter | null>;
+    publicKey: Ref<PublicKey | null>;
+    ready: Ref<boolean>;
+    connected: Ref<boolean>;
+    connecting: Ref<boolean>;
+    disconnecting: Ref<boolean>;
+
+    // Methods.
+    select(walletName: WalletName): void;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    sendTransaction(transaction: Transaction, connection: Connection, options?: SendTransactionOptions): Promise<TransactionSignature>;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transaction: Transaction[]): Promise<Transaction[]>;
+    signMessage(message: Uint8Array): Promise<Uint8Array>;
+}
+
+export interface WalletStoreProps {
+    wallets: Wallet[];
+    autoConnect?: boolean;
+    onError?: (error: WalletError) => void;
+    localStorageKey?: string;
+}
+
+let walletStore: WalletStore = {} as WalletStore;
+
+export const useWallet = (): WalletStore => walletStore;
+
+export const initWallet = ({
+    wallets,
+    autoConnect = false,
+    onError = (error: WalletError) => console.error(error),
+    localStorageKey = 'walletName',
+}: WalletStoreProps): void => {
+    const walletName = useLocalStorage<string>(localStorageKey);
+    const wallet = ref<Wallet | null>(null);
+    const adapter = ref<Adapter | null>(null);
+    const publicKey = ref<PublicKey | null>(null);
+    const ready = ref<boolean>(false);
+    const connected = ref<boolean>(false);
+    const connecting = ref<boolean>(false);
+    const disconnecting = ref<boolean>(false);
+
+    const walletsByName = computed<WalletDictionary>(() => {
+        return wallets.reduce((walletsByName, wallet) => {
+            walletsByName[wallet.name] = wallet;
+            return walletsByName;
+        }, {} as WalletDictionary);
+    });
+
+    // Update the wallet and adapter based on the wallet provider.
+    watch(walletName, (): void => {
+        wallet.value = walletsByName.value?.[walletName.value as string] ?? null;
+        adapter.value = wallet.value?.adapter() ?? null;
+        if (adapter.value) {
+            ready.value = adapter.value.ready;
+            publicKey.value = adapter.value.publicKey;
+            connected.value = adapter.value.connected;
+        } else {
+            ready.value = false;
+            publicKey.value = null;
+            connected.value = false;
+        }
+    }, { immediate:true });
```

Comment:
Will you explain what this flag does for me?

Reply:
In Vue:

- `watchEffect(fn)` is the equivalent of `useEffect(fn)` but you cannot provide a list of variables to listen for, instead it will automatically recognise any reactive variables inside the function and call that method every time one or more of these variables change during a given tick. Most of the time, this is exactly what we need and so we let Vue handle the list of vars for us.
- If you want to explicitly tell Vue what you're watching, you can use `watch(var, fn)` or `watch([vars], fn)`.
- However, by default, the `watch` method will only be called on future updates and not immediately after defining it (as `watchEffect` and `useEffect` do).
- Therefore, you need to provide that flag to tell it to trigger immediately.
- In conclusion, `watch([vars], fn, { immediate:true })` is equivalent to `useEffect(fn, [vars])`.

Reply:
Thanks for the detailed explanation!